AstSemantics.md

```diff
@@ -313,43 +355,27 @@ explicit accesses to linear memory.
 restriction may be lifted in the future.
 
-Direct calls to a function specify the callee by index into a *main function table*.
+Direct calls to a function specify the callee by index into the 
+[function index space](Modules.md#index-spaces).
 
   * `call`: call function directly
 
 A direct call to a function with a mismatched signature is a module verification error.
 
-Like direct calls, calls to [imports](Modules.md#imports-and-exports) specify
-the callee by index into an *imported function table* defined by the sequence of import
-declarations in the module import section. A direct call to an imported function with a
-mismatched signature is a module verification error.
-
-  * `call_import` : call imported function directly
-
-Indirect calls allow calling target functions that are unknown at compile time.
-The target function is an expression of value type `i32` and is always the first
-input into the indirect call.
-
-A `call_indirect` specifies the *expected* signature of the target function with
-an index into a *signature table* defined by the module. An indirect call to a
-function with a mismatched signature causes a trap.
-
-  * `call_indirect`: call function indirectly
-
-Functions from the main function table are made addressable by defining an
-*indirect function table* that consists of a sequence of indices into the
-module's main function table. A function from the main table may appear more
-than once in the indirect function table. Functions not appearing in the
-indirect function table cannot be called indirectly.
-
-In the MVP, indices into the indirect function table are local to a single
-module, so wasm modules may use `i32` constants to refer to entries in their own
-indirect function table. The [dynamic linking](DynamicLinking.md) feature is
-necessary for two modules to pass function pointers back and forth. This will
-mean concatenating indirect function tables and adding an operator `address_of`
-that computes the absolute index into the concatenated table from an index in a
-module's local indirect table. JITing may also mean appending more functions to
-the end of the indirect function table.
+Indirect calls to a function indicate the callee with an `i32` index into
+the [table](#tables) specified by a [table index](BinaryEncoding.md#index-spaces)
+immediate. The *expected* signature of the target function (specified by its
+index in the [types section](BinaryEncoding.md#type-section)) is given as a
+second immediate. 
+
+  * `call_table`: call function element in a table
+
+The specified [table](#tables) must have either a "function" or "function with
+signature `X`" element type (in the MVP, these are the only two
+possibilities, but in the future, there may be other kinds of tables).
+If the table element type has a signature, it must match the signature of the
+call. Otherwise, the signature is checked at runtime and a signature mismatch
+causes a trap.
```

Comment:
@jfbastien Pushed some clarifications, ptal.
@JSStats Agreed and great!  Feel free to delete (via 'X') "resolved" comments since I periodically scan open comments.